espnet/nets/pytorch_backend/e2e_tts_fastspeech.py

```diff
@@ -0,0 +1,686 @@
+                assert n1 == n2, "It seems that encoder structure is different."
+                assert p1.shape == p2.shape, "It seems that encoder size is different."
+                p1.data.copy_(p2.data)
+        elif init_module == "embed":
+            student_shape = self.encoder.embed[0].weight.data.shape
+            teacher_shape = self.teacher.encoder.embed[0].weight.data.shape
+            assert student_shape == teacher_shape, "It seems that embed dimension is different."
+            self.encoder.embed[0].weight.data.copy_(
+                self.teacher.encoder.embed[0].weight.data)
+        else:
+            raise NotImplementedError("Support only all or embed.")
+
+    def _reset_parameters(self, args):
+        if self.use_scaled_pos_enc:
+            # alpha in scaled positional encoding init
+            self.encoder.embed[-1].alpha.data = torch.tensor(args.initial_encoder_alpha)
+            self.decoder.embed[-1].alpha.data = torch.tensor(args.initial_decoder_alpha)
+
+        if args.transformer_init == "pytorch":
+            return
+        # weight init
+        for p in self.parameters():
+            if p.dim() > 1:
+                if args.transformer_init == "xavier_uniform":
+                    torch.nn.init.xavier_uniform_(p.data)
+                elif args.transformer_init == "xavier_normal":
+                    torch.nn.init.xavier_normal_(p.data)
+                elif args.transformer_init == "kaiming_uniform":
+                    torch.nn.init.kaiming_uniform_(p.data, nonlinearity="relu")
+                elif args.transformer_init == "kaiming_normal":
+                    torch.nn.init.kaiming_normal_(p.data, nonlinearity="relu")
+                else:
+                    raise ValueError("Unknown initialization: " + args.transformer_init)
+        # bias init
+        for p in self.parameters():
+            if p.dim() == 1:
+                p.data.zero_()
+        # reset some modules with default init
+        for m in self.modules():
+            if isinstance(m, (torch.nn.Embedding, LayerNorm)):
+                m.reset_parameters()
+
+    @property
+    def attention_plot_class(self):
+        return TTSPlot
+
+    @property
+    def base_plot_keys(self):
+        """base key names to plot during training. keys should match what `chainer.reporter` reports
+
+        if you add the key `loss`, the reporter will report `main/loss` and `validation/main/loss` values.
+        also `loss.png` will be created as a figure visulizing `main/loss` and `validation/main/loss` values.
+
+        :rtype list[str] plot_keys: base keys to plot during training
+        """
+        plot_keys = ["loss", "l1_loss", "duration_loss"]
+        if self.use_scaled_pos_enc:
+            plot_keys += ["encoder_alpha", "decoder_alpha"]
+
+        return plot_keys
+
+
+class DurationPredictorLoss(torch.nn.Module):
```

Comment:
this module has too many classes. please split these into new modules like `espnet.nets.pytorch_backend.fast_speech.dulation_predictor_loss`. remember @gtache's advice.

Reply:
fixed.